library/src/main/kotlin/io/intrepid/bleidiom/BleIdiomDevice.kt

```diff
@@ -107,51 +105,37 @@ class BleIdiomDevice internal constructor(internal val device: RxBleDevice) {
 
     override fun toString() = printName
 
-    private fun createConnection(): Observable<BleConnection> =
-            establishConnection()
-                    .flatMap {
-                        when (it) {
-                            is Try.Success -> Observable.just(it)
-                            is Try.Failure -> handleError(it)
-                        }
-                    }
+    private fun createConnection(): Observable<BleConnection> {
+        val establishedConnection = Observable.defer { device.establishConnection(autoConnect) }
```

Comment:
Whenever a brand-new first subscription is made, call `device.estalishConnection(autoConnect)`. Subsequent subscriptions, before the resulting Rx chain is finalized, are shared through the `ReplayingUnfinalizedShare`.